src/FakeItEasy/Expressions/ArgumentConstraints/EqualityArgumentConstraint.cs

```diff
@@ -32,7 +36,18 @@ public override string ToString()
                 return "\"{0}\"".FormatInvariant(stringValue);
             }
 
-            return this.ExpectedValue.ToString();
+            try
+            {
+                return this.ExpectedValue.ToString();
+            }
+            catch (Exception)
+            {
+                var type = this.ExpectedValue.GetType();
+
+                return type.Namespace == "Castle.Proxies"
+                    ? $"Faked {type.GetTypeInfo().BaseType.FullName}"
+                    : type.Name;

```

Comment:
Non-Fakes isn't quite just the name. It was the lack of namespace that tipped me off:

`System.Console.WriteLine((new List<int>()).GetType().Name);`

List`1

> System.Console.WriteLine((new List<int>()).GetType().FullName);

System.Collections.Generic.List`1[[System.Int32, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]

So just outputting `Name` still doesn't match the `ToString`, but it's perhaps a reasonable compromise.

The deeper issue, I suppose, why Fakes describe themselves differently. Maybe they shouldn't. Probably it doesn't matter. Either way, we don't need to address that question in this issue.

Thanks!